spring-integration-feed/src/main/java/org/springframework/integration/feed/inbound/FeedEntryMessageSource.java

```diff
@@ -206,7 +210,7 @@ private SyndEntry getNextEntry() {
 		else {
 			this.lastTime += 1; //NOSONAR - single poller thread
 		}
-		this.metadataStore.put(this.metadataKey, this.lastTime + "");
+		Objects.requireNonNull(this.metadataStore).put(this.metadataKey, this.lastTime + "");
```

Comment:
We prefer to avoid `Objects.requireNonNull` as much as possible, especially in critical path executions.
So, use `@SuppressWarnings("NullAway.Init")` on the `metadataStore` property instead.
In fact, it is really initialized from the `onInit()`.